py-rse/git-cmdline.Rmd

```diff
@@ -230,31 +315,85 @@ so that we can write a longer [commit message][commit-message].
 Once again,
 we put the message in double quotes because it contains spaces.
 
-If we run `git status` now,
-Git tells us that everything is up to date:
+If we run `git status` now:
+```shell
+$ git status
+```
+
+```text
+On branch master
+nothing to commit, working tree clean
+```
+Git tells us all of our existing work is tracked and up to date.  
+
+This first commit becomes the first point in our Git history --- 
+you can't time travel back beyond this point. 
+This means you generally want to initialize your project as a Git repository sooner rather than later.
+
+## How can I save and track changes? {#py-rse-git-cmdline-changes}
+
+Our initial commit gives us a starting point in our history.
+Now, let's see how we might record the history of a new file. 
+The process is similar to adding our existing work: 
+first add the file, then commit changes.
+
+Again, verify you are still in the right directory:
+
+```shell
+$ cd ~/zipfs-law
+```
+
+Let's use our `plotcounts.py` program to generate 
+a plot of the word counts in `results/anne_of_green_gables.csv`:
+```shell
+python bin/plotcounts.py results/anne_of_green_gables.csv --outfile results/anne_of_green_gables.png
+```
 
+We can check the status of our repo again:
 ```shell
 $ git status
 ```
 
 ```text
 On branch master
-nothing to commit, working directory clean
+Untracked files:
+  (use "git add <file>..." to include in what will be committed)
+
+        results/anne_of_green_gables.png
+
+nothing added to commit but untracked files present (use "git add" to track)
+```
+
+Since `anne_of_green_gables.png` is a new file, git isn't tracking it yet.
```

Comment:
Is it common practice to version/track .png files? 

If not (and since `anne_of_green_gables.csv` isn't generated in the scripting lesson), perhaps we could create, add and commit a new `anne_of_green_gables.csv` file here instead of the png file?